File: SmartApp/Shared.Models/Settings/EmailCleanerSettings/EmailCleanerConfiguration.cs
namespace Shared.Models.Settings.EmailCleanerSettings
{
    public class EmailCleanerConfiguration
    {
        public int UserId { get; set; }
        public List<EmailCleanerAccount> Accounts { get; set; } = new List<EmailCleanerAccount>();
        public List<EmailCleanerSettings> Settings { get; set; } = new List<EmailCleanerSettings>(); 
    }
}

File: SmartApp/Web.React/web-app/src/Pages/Settings/EmailCleanerSettings/Types/EmailCleanerConfiguration.ts
export type EmailCleanerConfiguration = {
  userId: number;
  accounts: EmailCleanerAccount[];
  settings: EmailCleanerSettings[];
};

export type EmailCleanerAccount = {
  id: number;
  accountName: string;
};

export type EmailCleanerSettings = {
  settingsId: number;
  enabled: boolean;
  allowReadEmails: boolean;
  allowMoveEmails: boolean;
  allowDeleteEmails: boolean;
  allowCreateEmailFolder: boolean;
  shareEmailDataToTrainAi: boolean;
  scheduleCleanup: boolean;
  scheduleCleanupAtHour: number;
  hasMappings: boolean;
  lastCleanupTime: string;
  mappings: EmailAddressMapping[];
};

export type EmailAddressMapping = {
  sourceAddress: string;
  domain: string;
  shouldCleanup: boolean;
  isSpam: boolean;
  predictedAs: string;
};
